common/mutex.h

```diff
@@ -79,6 +79,20 @@ class Mutex {
 
 /** @} */
 
+#else
+
+struct StackLock {
+	explicit StackLock(MutexInternal *mutex, const char *mutexName = nullptr) {}
+	explicit StackLock(const Mutex &mutex, const char *mutexName = nullptr) {}
+};
+
+struct Mutex {
```

Comment:
These should probably remain as classes, since some compilers warn about forward declarations. Using `final` and `inline` keywords may also help.

Reply:
AFAIK `inline` is applied automatically for methods defined in header file. `final` can't hurt, indeed.